UninitializedPropertyAccessException with slf4j v2.0.17

Hello. Slf4j v2.0.17 was released just recently. 

I've upgraded to it and got

> kotlin.UninitializedPropertyAccessException: lateinit property mdcAdapter has not been initialized

To reproduce this, you will only need `org.slf4j:slf4j-api:2.0.17` and `io.klogging:slf4j-klogging:0.9.2` in classpath:

```kotlin
plugins {
    java
}

repositories {
    mavenCentral()
}

dependencies {
    implementation("org.slf4j:slf4j-api:2.0.17")
    implementation("io.klogging:slf4j-klogging:0.9.2")
}
```

The startup of this minimal Java app will fail (it's enough to only have static logger field in the class):

```java
public class App {
    private static Logger logger = LoggerFactory.getLogger(App.class);

    public static void main(String[] args) {}
}
```

with the following exception:

<details>
  <summary>Click to expand</summary>

```plaintext
SLF4J(E): Failed to instantiate SLF4J LoggerFactory
SLF4J(E): Reported exception:
kotlin.UninitializedPropertyAccessException: lateinit property mdcAdapter has not been initialized
	at io.klogging.slf4j.KloggingServiceProvider.getMDCAdapter(KloggingServiceProvider.kt:44)
	at org.slf4j.LoggerFactory.earlyBindMDCAdapter(LoggerFactory.java:227)
	at org.slf4j.LoggerFactory.bind(LoggerFactory.java:199)
	at org.slf4j.LoggerFactory.performInitialization(LoggerFactory.java:187)
	at org.slf4j.LoggerFactory.getProvider(LoggerFactory.java:511)
	at org.slf4j.LoggerFactory.getILoggerFactory(LoggerFactory.java:497)
	at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:446)
	at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:472)
	at org.example.App.<clinit>(App.java:7)
Exception in thread "main" java.lang.ExceptionInInitializerError
Caused by: java.lang.IllegalStateException: Unexpected initialization failure
	at org.slf4j.LoggerFactory.bind(LoggerFactory.java:216)
	at org.slf4j.LoggerFactory.performInitialization(LoggerFactory.java:187)
	at org.slf4j.LoggerFactory.getProvider(LoggerFactory.java:511)
	at org.slf4j.LoggerFactory.getILoggerFactory(LoggerFactory.java:497)
	at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:446)
	at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:472)
Caused by: java.lang.IllegalStateException: Unexpected initialization failure

	at org.example.App.<clinit>(App.java:7)
Caused by: kotlin.UninitializedPropertyAccessException: lateinit property mdcAdapter has not been initialized
	at io.klogging.slf4j.KloggingServiceProvider.getMDCAdapter(KloggingServiceProvider.kt:44)
	at org.slf4j.LoggerFactory.earlyBindMDCAdapter(LoggerFactory.java:227)
	at org.slf4j.LoggerFactory.bind(LoggerFactory.java:199)
	... 6 more
```
  
</details>

--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -11,7 +11,7 @@ kotlin-coroutines = "1.10.1"
 kotlin-datetime = "0.6.2"
 kotlin-serialisation = "1.8.0"
 reckon = "0.19.1"
-slf4j = "2.0.16"
+slf4j = "2.0.17"
 spotless = "7.0.2"
 test-logger = "3.2.0"
 android-library = { strictly = "[8.7, 8.8[" }

--- a/slf4j-klogging/src/main/kotlin/io/klogging/slf4j/KloggingServiceProvider.kt
+++ b/slf4j-klogging/src/main/kotlin/io/klogging/slf4j/KloggingServiceProvider.kt
@@ -35,7 +35,7 @@ public const val REQUESTED_API_VERSION: String = "2.0.99"
 public class KloggingServiceProvider : SLF4JServiceProvider {
     private lateinit var loggerFactory: ILoggerFactory
     private lateinit var markerFactory: IMarkerFactory
-    private lateinit var mdcAdapter: MDCAdapter
+    private val mdcAdapter: MDCAdapter = BasicMDCAdapter()
 
     override fun getLoggerFactory(): ILoggerFactory = loggerFactory
 
@@ -48,7 +48,6 @@ public class KloggingServiceProvider : SLF4JServiceProvider {
     override fun initialize() {
         loggerFactory = NoCoLoggerFactory()
         markerFactory = BasicMarkerFactory()
-        mdcAdapter = BasicMDCAdapter()
 
         // Ensure any MDC items are included in every log event, whether an `NoCoLoggerWrapper` or not.
         Context.addItemExtractor {

